alts/src/main/java/io/grpc/alts/CheckGcpEnvironment.java

```diff
@@ -63,13 +63,15 @@ static boolean checkBiosDataOnWindows(BufferedReader reader) throws IOException
     }
     return false;
   }
+  
+  private static final String OS = System.getProperty("os.name").toLowerCase(Locale.ENGLISH);
 
   private static boolean isRunningOnGcp() {
     try {
-      if (SystemUtils.IS_OS_LINUX) {
+      if (OS.indexOf("linux") >= 0) {
         // Checks GCE residency on Linux platform.
         return checkProductNameOnLinux(Files.newBufferedReader(Paths.get(DMI_PRODUCT_NAME), UTF_8));
-      } else if (SystemUtils.IS_OS_WINDOWS) {
+      } else if (OS.indexOf("win") >= 0) {
```

Comment:
How did you come up with this code? Why not the more conventional `startsWith()` and why not use "windows" here?

Reply:
![image](https://user-images.githubusercontent.com/1005544/103024737-5327a100-4548-11eb-9864-76cf94ab6c3d.png)

I picked "win" instead of "windows" in case we end up with something weird like "Win95" or "WinXP" one of these days. indexOf seemed a little more general than startsWith.

Reply:
But it seems you copied from a lower-ranked answer that initially had bugs like "checking for 'darwin' after 'win'" if you look at the very first comment. And you can't simply remove the Darwin case and expect proper behavior.

OpenJDK uses [very regular names for Windows](https://github.com/openjdk/jdk/blob/91244cc738e92163d99b8951c5d95b546447f341/src/java.base/windows/native/libjava/java_props_md.c#L478). Mac is hard-coded to "Mac OS X" and Unix uses `uname -s`. I don't see the hard-coding for OS X in Java 7, so it probably used "Darwin." That's all obviously OpenJDK-specific, but I'd expect other JDKs to be similar (use uname for most cases, start Windows names with "Windows").

Use `startsWith()` and spell out "windows" in full.

Reply:
done